### PR TITLE
Add DU ecosystem to supported contracts

### PR DESCRIPTION
**Pull request by dogsunchained**
Added support for:
- Dogs Unchained
- Proof of Steak
- Puppies Unchained

Each token type requires a single call to an info contract to get ownership for a given address.

### CODE REVIEW

File: src/adapters/dogs-unchained.ts [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+    'Puppies Unchained': 9999,
+  };
+
+  private async getOwnedAssets(
+    contract: DuStaking,
+    asset: StakingAsset,
+    owner: string,
+    maxId: number,
+  ) {
+    const token = asset.asset.replace('ERC721:', '');
```
Comment (raymondfeng) [inline src/adapters/dogs-unchained.ts:50]: You can use the code below to pass `ERC721:<address>` as Chain Agnostic ID.

```ts
import {AssetName} from '@collabland/chain';
const token = new AssetName(asset.asset).reference;
```

**Comment by raymondfeng** [comment]
@dogsunchained Kudos to the first contribution! 🥇

File: src/adapters/dogs-unchained.ts (hunk 1 above)
Comment (dogsunchained) [reply src/adapters/dogs-unchained.ts]: Updated! Heads up that tests passed on my end (checked linter as well), but looks like CI/CD is erroring on a missing key of some sort? Let me know if I can help resolve.

Comment (raymondfeng) [reply src/adapters/dogs-unchained.ts]: Yes, `INFURA_PROJECT_ID` is configured as a github secret and it's invisible to pull requests.

**Comment by raymondfeng** [comment]
Please squash the commits and sign off using https://git-scm.com/docs/git-commit#Documentation/git-commit.txt--s

**Comment by dogsunchained** [comment]
> Please squash the commits and sign off using https://git-scm.com/docs/git-commit#Documentation/git-commit.txt--s

Hope I got this right! Let me know.

**Review by raymondfeng — APPROVED** [review]

**Comment by dogsunchained** [comment]
Thanks for the approval @raymondfeng! How can we start utilizing this in CollabLand?

**Comment by raymondfeng** [comment]
When the release is picked up by Collab.Land, your contract will be shown up in the Command Center when you select `Staking` as the token type to define token granted roles.

**Comment by dogsunchained** [comment]
Awesome. How often do they pick up these releases?

**Pull request merged**